manager/manager.go

```
@@ -973,6 +1024,46 @@ func (m *manager) populateBundleInput(bs model.Service, in *model.ServiceDataIte
 	return nil
 }
 
+func (m *manager) processSecret(svcName, itemName string, v *model.ServiceDataItem, val interface{}) error {
+	var b []byte
+	var err error
+	if s, ok := val.(string); ok { // if numeric type
+		b = []byte(s)
+	} else {
+		b, err = json.Marshal(val)
+	}
+
+	if err != nil {
+		return err
+	}
+	_, id, err := m.assets.Add(fmt.Sprintf("%s.%s", svcName, itemName), b, 1)
+	if err != nil {
+		return err // TODO: robustify
+	}
+	// TODO: Use configured URL.
+	v.Actual = m.secretURLFromToken(id)
+	v.SecretSet = true
+	v.Set = true
+	v.Value = v.Actual
+	return nil
+}
+
+func (m *manager) secretURLFromToken(t string) string {
+	return fmt.Sprintf("%v/v1/secrets/assets/redeem/%s", m.tasker.APIURL(), t)
+}
+
+func tokenFromURL(a interface{}) (string, bool) {
+	s, ok := a.(string)
+	if !ok {
+		return "", false
+	}
+	parts := strings.Split(s, "/")
+	if len(parts) < 3 || parts[len(parts)-2] != "redeem" {
+		return "", false
```

feel like `tokenFromUrl` should return an error instead of using the bool.